### DISCUSSION TITLE
Multiple Routes per Template

### DISCUSSION THREAD

**Discussion by acalvino4** [ExpressionEngine/ExpressionEngine#3312] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
It would be great if we could map multiple routes to use a single template. There are some fancy ways you could use regex or optional segments to achieve this in some cases, but it doesn't work for example, if I wanted a template to be used for both `/` and `/page-name`.

**Describe the solution you'd like**
An ideal user experience would be the ability to add the same template twice in the route manager, or to be able to add multiple entries to template route override on the template access page. Even a coma separate list in the same field would be fine.

**Describe alternatives you've considered**
Currently the best alternative I am aware of is to create a second template that's exactly the same but with a different route. It's just not an ideal user experience. And I know you can move all the common template to an embed (or layout), and then just include that in both templates, but again....just not ideal.

**Teachability, Documentation, Adoption, Migration Strategy**
From a teachability standpoint, I think there would be less to teach since it would be more intuitive than the current workarounds.

- [x] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by jcogs-design** [comment]
The easiest way to do this is to use the [Template Layout](https://docs.expressionengine.com/latest/templates/layouts.html#template-layouts) system.
Simply put one tag at the start of the 'duplicate' template and your done.
The processing overhead is minimal.
So for your example, put this at the top of the `index.html` template in your default template group
```
{layout='page-name'}
```
Then create your desired shared template as the index.html template in the template group `page-name`.

**Comment by acalvino4** [comment]
@jcogs-design thanks! this is essentially what we've done, so I am aware of the workarounds....but I think managing this through a routing system would be cleaner and more user friendly.

**Comment by acalvino4** [comment]
Also of note, I found another (limited) workaround!
Because my use case was matching `/` and `/{page:alpha_dash}`, I used the `/site/index` template and set the override to `/{page:alpha_dash}`. Because `/site/index` is loaded by default for the empty route `/`, this works for me.

**Comment by intoeetive** [comment]
I don't know if it still works with latest EE, but also have a look at https://github.com/rsanchez/resource_router

**Comment by robsonsobral** [comment]
Maybe this "advanced" option could be set on `config.php`.

**Comment by acalvino4** [comment]
I didn't see any other open ticket directly about this issue, but I noticed the idea mentioned in this comment:
https://github.com/ExpressionEngine/ExpressionEngine/issues/548#issuecomment-785018800

I also agree with @intoeetive in linked comment that "ee routes need more improvements in general", but that could be another whole convo.

**Comment by intoeetive** [comment]
https://github.com/ExpressionEngine/ExpressionEngine/pull/4486 (needs to be tested)
